compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -525,6 +527,36 @@ public void visit(BLangIf ifNode) {
         this.uninitializedVars = mergeUninitializedVars(ifResult.uninitializedVars, elseResult.uninitializedVars);
     }
 
+    @Override
+    public void visit(BLangMatchStatement matchStatement) {
+        analyzeNode(matchStatement.expr, env);
+
+        Map<BSymbol, InitStatus> unInitVars = new HashMap<>();
+        BranchResult lastClauseResult = null;
+        for (BLangMatchClause matchClause : matchStatement.matchClauses) {
+            if (matchClause.isLastClause) {
+                lastClauseResult = analyzeBranch(matchClause, env);
+            } else {
+                BranchResult result = analyzeBranch(matchClause, env);
+                // If the flow was terminated within the block, then that branch should not be considered for
+                // analyzing the data-flow for the downstream code.
+                if (result.flowTerminated) {
+                    continue;
+                }
+                unInitVars = mergeUninitializedVars(unInitVars, result.uninitializedVars);
+            }
+        }
+
+        if (lastClauseResult != null) {
+            // only if last pattern is present, uninitializedVars should be updated
```

Comment:
What do we mean by last pattern and what does it mean to say there is no last pattern present?

Reply:
This part is taken from the previous implementation since this is handled in a different way in the new implementation, this code is deleted.